endpoint.go

```diff
@@ -349,44 +249,36 @@ func (ep *endpoint) joinLeaveEnd() {
 	}
 }
 
-func (ep *endpoint) Join(containerID string, options ...EndpointOption) error {
+func (ep *endpoint) Join(sbox Sandbox, options ...EndpointOption) error {
 	var err error
 
-	if containerID == "" {
-		return InvalidContainerIDError(containerID)
+	if sbox == nil {
+		return types.BadRequestErrorf("endpoint cannot be joined by nil container")
+	}
+
+	s, ok := sbox.(*sandboxData)
+	if !ok {
+		return types.BadRequestErrorf("not a valid Sandbox interface")
 	}
```

Comment:
in addition to all these checks, we must also lookup the sandboxtable to confirm the presence of the sandbox before proceeding.